src/main/scala/is/hail/methods/LinearMixedRegressionDist.scala

```diff
@@ -0,0 +1,216 @@
+package is.hail.methods
+
+import breeze.linalg._
+import is.hail.annotations._
+import is.hail.distributedmatrix.DistributedMatrix
+import is.hail.expr._
+import is.hail.stats._
+import is.hail.utils._
+import is.hail.variant.VariantDataset
+import org.apache.spark.mllib.linalg.{DenseMatrix => SparkDenseMatrix}
+import org.apache.spark.mllib.linalg.distributed.{BlockMatrix, IndexedRow}
+import is.hail.distributedmatrix.DistributedMatrix.implicits._
+
+object LinearMixedRegressionDist {
+  val dm = DistributedMatrix[BlockMatrix]
+
+  import dm.ops._
+
+  def applyEigenDist(
+    vds: VariantDataset,
+    eigenDist: EigendecompositionDist,
+    yExpr: String,
+    covExpr: Array[String],
+    useML: Boolean,
+    rootGA: String,
+    rootVA: String,
+    runAssoc: Boolean,
+    optDelta: Option[Double],
+    sparsityThreshold: Double,
+    useDosages: Boolean): VariantDataset = {
+
+    require(vds.wasSplit)
+
+    val pathVA = Parser.parseAnnotationRoot(rootVA, Annotation.VARIANT_HEAD)
+    Parser.validateAnnotationRoot(rootGA, Annotation.GLOBAL_HEAD)
+
+    val (y, cov, completeSamples) = RegressionUtils.getPhenoCovCompleteSamples(vds, yExpr, covExpr)
+    val C = cov
+    val completeSamplesSet = completeSamples.toSet
+    val sampleMask = vds.sampleIds.map(completeSamplesSet).toArray
```

Comment:
As I said on regular LMMReg, seems this stuff is only necessary if runAssoc is true. Maybe move it down?